### DISCUSSION TITLE
filter prefixes on utilization

### DISCUSSION THREAD

**Discussion by PieterL75** [netbox-community/netbox#6605] [Q&A]
Hey,

Is there a way to filter out '100%' used prefixes from the results (using custom scripts ObjectVar)?
I dont's see a property in the API, so assume the number is calculated on the fly when displaying the tables in the UI.

It would allow me to show only prefixes that have room in a dropdown of my custom scripts.

Pieter

**Comment by jeremystretch** [comment]
> I dont's see a property in the API, so assume the number is calculated on the fly when displaying the tables in the UI.

Correct. Unfortunately this precludes us from filtering on utilization, because it's not calculated until after the database query has been made. This could be changed, as we did with storing prefix depth and child count in #6087, but AFAIK no proposal has been made yet.

**Reply by PieterL75** [reply]
I created #6606 as a new feature request for this.

Appreciated !

Pieter

**Comment by candlerb** [comment]
For right now you might consider adding a tag to 'full' prefixes (and keeping it up to date with a webhook).  However I'm not sure you can filter on the absence of a tag, so you might have to tag 'available' prefixes instead.

**Reply by PieterL75** [reply]
a 'fulltag' seems to be a problem, as I can't find a way to filter out prefixes that do NOT have a tag...

I ended up adding a CustomField that populates the utilization.. 
Little script that runs daily (or I might add a webhook to run the script trough an API call)

`class UpdatePrefixUsageCF(Script):
    class Meta:
        name = "Set the % usage of a Prefix in the CustomField 'Utilization'"
        commit_default = False

    def run(self, data, commit):
        for prefix in Prefix.objects.filter():
            util=prefix.get_utilization()
            if not 'Utilization' in prefix.cf.keys():
                prefix.cf['Utilization']=101
            if util!=prefix.cf['Utilization']:
                prefix.cf['Utilization']=util
                prefix.save()
                self.log_success(f"Prefix {prefix} utilization updated to {util}.")
`

I my enhancement request gets approved and validated, it's a small effort to change the field I compare to

The customfield solution, actually gave me the option to sort my prefixes on utilization now :-) nice side effect
